### PR TITLE
Retrigger builds on restart

### PR DESCRIPTION
**Pull request by sersorrel**
now we shouldn't lose progress on builds when g-s-b is restarted!

Depends on https://github.com/wtsi-hgi/softpack-core/pull/48.

### CODE REVIEW

**Review by mjkw31 — DISMISSED** [review]
LGTM.

File: core/core.go [hunk 1]
```diff
@@ -0,0 +1,190 @@
...
+	if cr.Data.CreateEnvironment != nil {
+		if cr.Data.CreateEnvironment.TypeName != createMutationSuccess {
+			return internal.Error(cr.Data.CreateEnvironment.Message)
+		}
+	}
+
+	return nil
+}
+
+type ResendResponse struct {
```
Comment (sb10) [inline core/core.go:162]: Public methods here and below need documenting.

File: internal/wrmock/mock.go [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+	Cmd                   string
+	Fail                  bool
+	PollForStatusInterval time.Duration
+	JobDuration           time.Duration
+
+	sync.RWMutex
+	ReturnStatus wr.WRJobStatus
+}
+
+func NewMockWR(pollForStatusInterval, jobDuration time.Duration) *MockWR {
```
Comment (sb10) [inline internal/wrmock/mock.go:44]: Needs public docs

File: remove/remove.go [hunk 3]
```diff
@@ -86,7 +70,15 @@ func Remove(conf *config.Config, s3r s3Remover, envPath, version string) error {
 		return err
 	}
 
-	if err := removeEnvFromCore(conf, envPath+"-"+version); err != nil {
+	slog.Info(fmt.Sprintf("Removing env %s from core\n", envPath))
```
Comment (sb10) [inline remove/remove.go:73]: lowercase first letter in logs messages

File: server/server.go [hunk 4]
```diff
@@ -26,16 +26,24 @@ package server
...
 const (
-	endpointEnvs       = "/environments"
-	endpointEnvsBuild  = endpointEnvs + "/build"
-	endpointEnvsStatus = endpointEnvs + "/status"
+	endpointEnvs            = "/environments"
+	endpointEnvsBuild       = endpointEnvs + "/build"
+	endpointEnvsStatus      = endpointEnvs + "/status"
+	stopTimeout             = 10 * time.Second
+	readHeaderTimeout       = 20 * time.Second
+	waitUntilStartedTimeout = 10 * time.Second
```
Comment (sb10) [inline server/server.go:46]: Increase this timeout, since apparently the re-send can seem quite slow?

File: server/server.go [hunk 5]
```diff
@@ -55,30 +63,124 @@ type Builder interface {
...
+	}
+
+	err := s.c.ResendPendingBuilds()
+	close(s.startedCh)
+
+	return err
+}
+
+// WaitUntilStarted blocks until Start() brings up a listener and the core
+// service talk to us. Will only work if we were configured with core details
```
Comment (sb10) [inline server/server.go:155]: talks

**Review by sb10 — CHANGES_REQUESTED** [review]
Some missed docs

**Review by sb10 — APPROVED** [review]

**Pull request merged**